packages/xo-server/src/xapi/mixins/vm.js

```diff
@@ -82,7 +84,18 @@ export default {
     )
     $defer.onFailure(() => this.deleteVm(vmRef))
 
-    // TODO: copy BIOS strings?
+    let vm = await this._getOrWaitObject(vmRef)
```

Comment:
You don't need to wait for the record of the `VM` to do the second step. You can know the virtualization type from the template.